chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -241,26 +213,27 @@ impl Actor for PeerManagerActor {
         if let Some(server_addr) = self.config.node_addr {
             debug!(target: "network", at = ?server_addr, "starting public server");
             let peer_manager_addr = ctx.address();
+            let event_sink = self.event_sink.clone();
 
             actix::spawn(async move {
-                match TcpListener::bind(server_addr).await {
-                    Ok(listener) => loop {
-                        if let Ok((conn, client_addr)) = listener.accept().await {
-                            peer_manager_addr.do_send(
-                                PeerManagerMessageRequest::InboundTcpConnect(
-                                    InboundTcpConnect::new(conn),
-                                ),
-                            );
-                            debug!(target: "network", from = ?client_addr, "got new connection");
-                        }
-                    },
+                let listener = match TcpListener::bind(server_addr).await {
+                    Ok(l) => l,
                     Err(e) => {
                         panic!(
                             "failed to start listening on server_addr={:?} e={:?}",
                             server_addr, e
                         );
                     }
                 };
+                event_sink.push(Event::ServerStarted);
```

Comment:
I also considered that. What stopped me was the fact that then port selection would have to happen:
- outside of constructor
- in an asynchronous context (which would require making all functions calling this constructor async as well
- it would require interpreting the network config, which should be responsibility of PeerManagerActor.

This is a lot of drawbacks which I do not have a full solution for.
I'd like to make PeerManagerActor constructor async, add a way to specify in networkconfig that port should be arbitrary, and expose the allocated port.

Reply:
closing, will address in another PR.